core/client/db/src/cache/list_cache.rs

```diff
@@ -512,6 +566,44 @@ impl<Block: BlockT, T: CacheItemT> Fork<Block, T> {
 			best_finalized_block,
 		)
 	}
+
+	/// Truncate fork by deleting all entries that are descendants of given block.
+	#[cfg(test)]
```

Comment:
What's wrong with having this available all the time?

I'd be in favour of having a function that you're not supposed to call rather than having `#[cfg(test)]` everywhere.

```suggestion
```

Reply:
Initially I was agree with your suggestion, but as I wrote the code, I have had some strange feelings surrounding me. So I would prefer to revert this commit && leave `#[cfg(test)]` instead. But if you still have a strong feeling that last commit is OK, then let it be